File: resource/cpp/com/template/Rule.cpp
#include "Rule.h"
void Rule::read(ByteBuffer* buffer)
{
		this->level=buffer->readInt();
		this->role_level=buffer->readInt();
		short consume_goods_count=buffer->readShort();
		if(consume_goods_count>0)
		{
			for(int i=0;i<consume_goods_count;i++)
			{
				std::string tmp=buffer->readUTF();
				this->consume_goods.push_back(tmp);
			}
		}

		this->probability=buffer->readFloat();
		this->num=buffer->readInt();
}

File: resource/java/com/template/Rule.java
package com.template;
import java.io.DataInputStream;
public class Rule implements TemplateInterface
{
	public int level;
	public int role_level;
	public String[] consume_goods;
	public float probability;
	public int num;
	public void read(DataInputStream in)throws Exception
	{
		level=in.readInt();
		role_level=in.readInt();
		short consume_goods_count=in.readShort();
		if(consume_goods_count>0)
		{
			consume_goods=new String[consume_goods_count];
			for(int i=0;i<consume_goods_count;i++)
			{
				consume_goods[i]=in.readUTF();
			}
		}

		probability=in.readFloat();
		num=in.readInt();
	}
}
